### DISCUSSION TITLE
add integrations tests and the code coverage.

### DISCUSSION THREAD

**Discussion by Sonichigo** [Budibase/budibase#12555] [Contributing]
I was going through the codebase and pr, to see how ways to contribute. I noticed that there is a folder for [tests](https://github.com/Budibase/budibase/tree/master/packages/server/src/tests/utilities). I use a tool called Keploy to for testing and generating realistic data mocks, I think it would be a nice to have some more tests and a bit higher code coverage. Can i contirbute in this way?

**Comment by melohagan** [comment]
Hey @Sonichigo 

If there's gaps in test coverage we'd certainly be happy for you to fill them! 

We are however focused on using vitest for our unit testing: https://vitest.dev/

Also note that each package has its own set of tests.

**Comment by mike12345567** [comment]
Hi @Sonichigo - our test suites are split out across the different packages, we utilise Jest, vitest and a product by QA-wolf for the majority of our tests, we would not wish to add another testing tool as this would require a lot of extra maintenance.

Our overall coverage is already around 80%
![image](https://github.com/Budibase/budibase/assets/4407001/660127e3-eba0-4aff-a3cd-3dfd0eb9bfc6)

You can find our backend tests in any `.spec.js` or `.spec.ts` file - if you wish to contribute we can provide a list of areas which have slightly lower coverage that are in need of expanded Jest tests.

**Reply by Sonichigo** [reply]
Thanks for let me know! Can you provide me a list of other backend service areas where I can contribute, since recently through this i got to know about vitest and learning more about it.

**Reply by melohagan** [reply]
Hey @Sonichigo you can check out our contributing guide here: https://github.com/Budibase/budibase/blob/HEAD/docs/CONTRIBUTING.md
